game/src/contrib/systems/CollisionSystem.java

```diff
@@ -19,18 +20,51 @@ public CollisionSystem() {
         super(CollideComponent.class);
     }
 
+    /**
+     * Test every CollideEntity with each other CollideEntity for collision.
+     *
+     * <p>The collision check will be performed only once for a given tuple of entities. When entity
+     * A does collide with entity B it also means B collides with A.
+     */
     @Override
     public void execute() {
-        getEntityStream()
-                .flatMap(
-                        a ->
-                                getEntityStream()
-                                        .filter(b -> a.id() < b.id())
-                                        .map(b -> buildData(a, b)))
-                .forEach(this::onEnterLeaveCheck);
+        getEntityStream().flatMap(this::createDataPairs).forEach(this::onEnterLeaveCheck);
+    }
+
+    /**
+     * Create a stream of pairs of entities
+     *
+     * <p>Pair a given entity with every other entity with a higher id.
+     *
+     * @param a Entity which is the lower id partner
+     * @return the stream which contains every valid pair of Entities
+     */
+    private Stream<CollisionData> createDataPairs(Entity a) {
+        return getEntityStream().filter(b -> isIDSmallerThen(a, b)).map(b -> newDataPair(a, b));
+    }
+
+    /**
+     * Compares the id of the given Entities.
+     *
+     * <p>Makes sure only Entities with a higher id are paired up. This Prevents a pair of Entities
+     * to be checked double or with itself.
```

Comment:
```suggestion
     * <p>This comparison is applied in the {@link #createDataPairs(Entity a) createDataPairs} method to create only tuples with entities with higher ID. This avoids performing a collision check twice for a pair of entities, first for (a,b) and second for (b,a).
```

Reply:
noch nicht formatiert, nur erstmal als idee für die formulierung.

Reply:
Klingt für mich besser. Vielleicht noch erwähnen, dass auch für identisch ist, abgesichert also `for (a==b)` aber das ist mein kleinlich sein wohl eher

Reply:
Theoretisch impliziert mit `create only tuples with entities with higher ID`

Reply:
ja, steckt da drin ("higher"). aber es schadet auch nicht, es explizit als beispiel mit aufzunehmen, dann ist es offensichtlicher :)